src/proxy/http/profiles/router.rs

```diff
@@ -0,0 +1,367 @@
+use futures::{Async, Poll, Stream};
+use http;
+use std::hash::Hash;
+
+use never::Never;
+
+use dns;
+use proxy::Error;
+use svc;
+
+use super::recognize::*;
+use super::*;
```

Comment:
Similarly, we prefer to avoid glob imports for module code.

Also, I'd avoid so many newlines between import groups. It's fine to have one big group, especially if crate-local imports are e.g. `use crate::dns;`